.github/workflows/codeql.yml

```diff
@@ -0,0 +1,77 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
```

Comment:
Sorry if this is a dummy question but just trying to understand our intention here. Why do we need to run this for push to branches as well as on a schedule? Wouldn't be enough if we only have it for pull requests?

Reply:
As stated in the comment in the yml:
```
https://github.com/aws/aws-sam-cli/pull/4552/checks?check_run_id=10533852983
```

I'll keep the on push as well, but limit the branches to only develop and main